src/stan/mcmc/windowed_adaptation.hpp

```diff
@@ -109,6 +109,25 @@ class windowed_adaptation : public base_adaptation {
     }
   }
 
+  /**
+   * Check if there is any more warmup left to do
+   */
+  bool finished() {
+    if (adapt_window_counter_ + 1 >= num_warmup_) {
```

Comment:
we can just return this expression instead of doing `if` clause, can we?

Reply:
Good point